### PR TITLE
Feature/benson272/983739318 item routes

### PR DESCRIPTION
**Pull request by Benson272**
### Administrative Info
Monday Board ID: 853628905
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Monday Item ID]-[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
What changes did you make?
- created routes for adding items and getting all items (localhost:9000/items/addItem and localhost:9000/items/getAllItems)

### Testing
How did you confirm your changes worked? 
- Sending POST and GET requests to add and get items from db
- Tested by adding items on same date (if added on same date, items should be combined into one batch)
- Tested by adding an item that does not already exist

### Confirmation of Change 
Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

| Screenshot of database after inserting items |
image

| Screenshot of api route |
image

| Screenshot of result from getAllItems route |
image

### CODE REVIEW

File: backend/routes/index.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
-const express = require('express');
+const express = require("express");
 const router = express.Router();
 
-router.get('/', function(req, res, next) {
-    res.send("Supply Chain");
+router.get("/", function (req, res, next) {
+  res.send("Sup.ply Chain");
```
Comment (justinyaodu) [inline backend/routes/index.js]: Is "Sup.ply" a typo?

File: backend/routes/items.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      // item does not exist, so create one and add the batch
+      let newItem = new Item({
+        name: itemName,
+        itemId: itemId,
+        batches: [batch],
+      });
+      newItem.save();
+    } else {
+      // if item already exist with same date, combine the two batches
+      for (let batch of item.batches) {
```
Comment (hahsu) [inline backend/routes/items.js:55]: Is this necessary? I thought we had talked about being able to use a property value to search through the array of batches. Maybe $elemMatch? [https://docs.mongodb.com/manual/reference/operator/query/elemMatch/](url)

**Review by justinyaodu — CHANGES_REQUESTED** [review]
I left a comment on one thing that might be a typo, and added `===` instead of `==` for a null check, but it looks ready to merge otherwise. Good stuff!

File: backend/routes/items.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    res.send(items, 200);
+  });
+});
+
+router.post("/addItem", (req, res, next) => {
+  // validate body params
+  const itemName = req.body["itemName"];
+  const itemId = req.body["itemId"];
+  const weight = req.body["weight"];
+  const today = toUTCMidnight(new Date());
```
Comment (hahsu) [inline backend/routes/items.js:20]: Is this getting today's date? Don't we want the expiration date here, which will be passed to you from the form? I guess depending on what design team answers regarding how the frontend calendar looks, we might need to keep track of what day the batch was added AND the expiration date. So maybe our batch needs the fields startDate and expirationDate. Can you update the model, and reflect that change in this file as well?

**Review by hahsu — CHANGES_REQUESTED** [review]

File: backend/routes/index.js (hunk 1 above)
Comment (petabite) [reply backend/routes/index.js]: Oops yup. Nice catch!

**Review by hahsu — APPROVED** [review]
lgtm

**Review by justinyaodu — APPROVED** [review]
LGTM!

**Pull request merged**